### ISSUE TITLE
Screenshot doesn't work when saving new simulation in ios 

### ISSUE THREAD

**Issue by simetenn** [CINPLA/neuronify#145]

**Comment by dragly** [comment]
@andeplane Does QtQuick Item.grabToImage work on iOS? I'm guessing we might be saving the image to a directory we don't have write permissions to (seems unlikely, as the simulation and screenshot is saved to StandardPaths.writableLocation(StandardPaths.AppConfigLocation, ...)) or that we are not allowed to take screenshots on iOS at all (DRM issue?).

http://doc.qt.io/qt-5/qml-qtquick-item.html#grabToImage-method

**Comment by andeplane** [comment]
@dragly yes it works. I created a working example [here](https://dl.dropboxusercontent.com/u/9786132/screenshottest.zip) (use the iOS simulator).

But it seems like when we use `QStandardPaths::AppConfigLocation`, the path becomes
`/Users/anderhaf/Library/Developer/CoreSimulator/Devices/898A2AFD-E1AB-4947-9219-AE5114790B59/data/Containers/Data/Application/802F9405-131A-4D15-81D7-433B84796D08/Library/Preferences/screenshottest/test.png`
where the deepest directory screenshottest is not created. If I create the directory it works. I suppose this is a bug in QStandardPath?

If we use `QStandardPaths::DocumentsLocation` it works with no modifications.

So we have two solutions:
1) Use another directory than `QStandardPaths::AppConfigLocation`
2) Create the folder if it does not exist

**Comment by andeplane** [comment]
We did some more testing yesterday. My example in the above comment doesn't work on the device (iOS 10). A .png is created, but it's empty. We don't know what this is yet.

**Comment by andeplane** [comment]
It could be related to [this bug](https://bugreports.qt.io/browse/QTBUG-45902), but I'm not sure.

**Comment by dragly** [comment]
Seems likely that it could be related to that bug. The patch mentioned in that bugreport didn't make it into Qt 5.7.0, but it is there in Qt 5.7.1 and 5.6.2:

In qtbase:
```
$ git checkout 5.6.2
$ git log | grep I900857cfa996924469aaaeeee8037dc84a4fd575
Change-Id: I900857cfa996924469aaaeeee8037dc84a4fd575
$ git checkout 5.7.0
$ git log | grep I900857cfa996924469aaaeeee8037dc84a4fd575
$ git checkout 5.7.1
$ git log | grep I900857cfa996924469aaaeeee8037dc84a4fd575
    Change-Id: I900857cfa996924469aaaeeee8037dc84a4fd575
```

Could you try to compile Neuronify with Qt 5.6.2? (Might have to compile and install QtCharts separately) Or with the latest Qt 5.7.1 snapshot? http://download.qt.io/snapshots/qt/5.7/5.7.1/491/

If not, or if it doesn't work, I think we just have to make a workaround on iOS where we show something like "Save 1", "Save 2", etc. instead of the screenshots.

With time, we should allow the user to save a name and description that could have been used, but this will require changing the file format and the UI so much that I would like to postpone this to v1.1.

**Comment by andeplane** [comment]
I tested with Qt 5.6.2 and screenshot capture works as expected. 5.7.1 crashed for some reason (I don't have macOS Sierra, so I can't debug on iOS 10, and simulator isn't shipped with 5.7.1 snapshot). 

I had to change to using quick controls 1 since 2.0 isn't supported before Qt 5.7.
